synapse/handlers/register.py

```diff
@@ -399,14 +399,14 @@ def get_or_create_user(self, localpart, displayname, duration_seconds):
 
             yield registered_user(self.distributor, user)
         else:
-            yield self.store.flush_user(user_id=user_id)
+            yield self.store.user_delete_access_tokens(user_id=user_id)
```

Comment:
If I understood wat you mean here, would the following doc work better?
    Creates a new user if the user does not exist but if the user exists it will replace the user's old access token with a new one.